cloudigrade/util/aws.py

```diff
@@ -122,49 +145,76 @@ def get_running_instances(arn):  # TODO filter
     return running_instances
 
 
-def get_assumed_session(arn, region_name='us-east-1', credentials=None):
+def verify_account_access(session):
     """
-    Return a session using the customer AWS account role.
+    Check role for proper access to AWS APIs.
 
     Args:
-        arn (str): Amazon Resource Name to use for assuming a role.
-        region_name (str): AWS Region to associate the session with.
-        credentials (dict): Credentials returned by STS assume_role call.
+        session (boto3.Session): A temporary session tied to a customer account
 
     Returns:
-        boto3.Session: A temporary session tied to a customer account
+        bool: Whether role is verified.
 
     """
-    if credentials is None:
-        credentials = get_credentials_for_arn(arn)
+    success = True
+    for action in cloudigrade_policy['Statement'][0]['Action']:
+        if not _verify_policy_action(session, action):
+            success = False
```

Comment:
You probably want `return False` here to break the iteration. Otherwise, if you have one action fail and the _next_ one succeeds, the whole function returns `True`.